_posts/2025-05-22-debezium-as-part-of-your-ai-solution.adoc

```diff
@@ -0,0 +1,438 @@
+            final var result = delegate.augment(augmentationRequest);
+            Log.infof("Result of augmentation is %s", result.contents());
+            return result;
+        }
+    }
+}
+----
+
+The `EmbeddingStoreContentRetriever` class is responsible for querying the Milvus vector database and enhances the prompt context with the three most similar documents.
+The class itself also provides the `RetrievalAugmentorDecorator`, which logs the detail of the augmentation operation.
+
+The Milvus embedding store is again configured in https://github.com/debezium/debezium-examples/blob/main/ai-rag/client/src/main/resources/application.properties[application.properties].
+
+----
+quarkus.langchain4j.milvus.host=localhost
+quarkus.langchain4j.milvus.port=19530
+quarkus.langchain4j.milvus.create-collection=false
+quarkus.langchain4j.milvus.collection-name=demo_ai_documents
+
+# These two settings are not supported by the current version of Quarkus Langchain4j extension
+# Also field with text cannot be overriden and is default text
+quarkus.langchain4j.milvus.primary-field=id
+quarkus.langchain4j.milvus.vector-field=vector
+----
+
+== Demo
+
+The demo is composed of two components
+
+* Docker Compose configuration https://github.com/debezium/debezium-examples/blob/main/ai-rag/docker-compose.yml[file] that starts all services.
+* Java CLI application in https://github.com/debezium/debezium-examples/tree/main/ai-rag/client[client] directory that emulates both data pipeline and AI application.
+
+=== Start deployment
+
+In *Terminal 1* execute the command and wait for the containers to start.
+
+----
+$ docker-compose up --build
+----
+
+The command will extend the standard Debezium Server image with the `debezium-ai-embeddings-ollama` module (the AI embeddings modules are not by default included).
+PostgreSQL (`postgres`), Milvus (`milvus`), Debezium Server (`debezium-server`) and ollama (`ollama`) containers are started.
+
+In *Terminal 2* execute this command to start the Granite LLM model.
+
+----
+$ docker exec -it ollama ollama run granite3.1-dense:2b
+----
+
+We need to make sure that we are testing the application with facts that the model is unaware of.
+First, we will check the model's cut-off date (what's the latest data the model has seen).
+
+----
+>>> What is your cut off date?
+My knowledge cut off date is April 2024. I can provide information and answer questions based on data up to 
+that point. For more recent events or discoveries, I'm unable to include those details in my responses as 
+they weren't available at the time of my training data. If you have any questions within this timeframe, 
+feel free to ask!
+----
+
+Based on that fact we selected paper https://arxiv.org/abs/2504.05309v1[2504.05309v1] (IterQR: An Iterative Framework for LLM-based Query Rewrite in e-Commercial Search System) as the sample of the new knowledge.
+It was published in February 2025 so after cut off date and contains very specific keyword - `IterQR`.
```

Comment:
maybe outline in one or two sentences what `IteQR` stands for (something line `enhancing e-commec search framework` would be enough) so the reader immediately knows how the initial answer of the app is completely unrelated to real term, without checking arxive article